### ISSUE TITLE
What is a visitor?

### ISSUE THREAD

**Issue by PedanticHacker** [niklasf/python-chess#657]
I have a question regarding visitors in `chess.pgn`. Can anyone explain to me, in layman terms, what a visitor is and what is its purpose?

**Comment by PedanticHacker** [comment]
Correct me if I'm wrong, but I think that a visitor is a way to determine exactly how the parser is going to parse PGNs. To define a visitor, one must subclass from the abstract base class `BaseVisitor()` and override its methods. And then to apply the visitor, one must call the `accept()` method of the `chess.pgn.Game()` class and pass that visitor to the `accept()` method.

Is my understanding correct?

**Comment by niklasf** [comment]
Correct. Custom visitors can also be used directly from `chess.pgn.read_game()`. One use-case is to parse games into a custom data structure instead of `chess.pgn.Game`.

**Comment by PedanticHacker** [comment]
Ah, thank you!
